tools/mason/MasonSearch.chpl

```diff
@@ -92,8 +89,19 @@ proc masonSearch(ref args: list(string)) {
       }
     }
   }
-
-  for r in results.toArray().sorted() do writeln(r);
+  // Sort the results in a order such that results that startWith needle
+  // are displayed first 
+  use Sort;
```

Comment:
In preparation of future changes (https://github.com/chapel-lang/chapel/issues/15715), how about we split off a function for ordering the results now, e.g.

```chpl
var res = rankResults(results); // <-- everything else takes place here.
for r in res do writeln(r);
```